src/OpenTelemetry.Api/Context/Propagation/AWSXRayPropagator.cs

```diff
@@ -0,0 +1,312 @@
+            {
+                OpenTelemetryApiEventSource.Log.FailedToExtractActivityContext(nameof(AWSXRayPropagator), "null carrier");
+                return context;
+            }
+
+            if (getter == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToExtractActivityContext(nameof(AWSXRayPropagator), "null getter");
+                return context;
+            }
+
+            try
+            {
+                var parentTraceHeader = getter(carrier, AWSXRayTraceHeaderKey);
+
+                if (parentTraceHeader == null || parentTraceHeader.Count() != 1)
+                {
+                    return context;
+                }
+
+                var parentHeader = parentTraceHeader.First();
+
+                if (!TryParseXRayTraceHeader(parentHeader, out var newActivityContext))
+                {
+                    return context;
+                }
+
+                return new PropagationContext(newActivityContext, context.Baggage);
+            }
+            catch (Exception ex)
+            {
+                OpenTelemetryApiEventSource.Log.ActivityContextExtractException(nameof(AWSXRayPropagator), ex);
+            }
+
+            return context;
+        }
+
+        /// <inheritdoc/>
+        public void Inject<T>(PropagationContext context, T carrier, Action<T, string, string> setter)
+        {
+            if (context.ActivityContext.TraceId == default || context.ActivityContext.SpanId == default)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "Invalid context");
+                return;
+            }
+
+            if (carrier == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "null carrier");
+                return;
+            }
+
+            if (setter == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "null setter");
+                return;
+            }
+
+            var sb = new StringBuilder();
+            sb.Append(RootKey);
+            sb.Append(KeyValueDelimiter);
+            sb.Append(ToXRayTraceIdFormat(context.ActivityContext.TraceId.ToHexString()));
+            sb.Append(TraceHeaderDelimiter);
+            sb.Append(ParentKey);
+            sb.Append(KeyValueDelimiter);
+            sb.Append(context.ActivityContext.SpanId.ToHexString());
+            sb.Append(TraceHeaderDelimiter);
+            sb.Append(SampledKey);
+            sb.Append(KeyValueDelimiter);
+            sb.Append((context.ActivityContext.TraceFlags & ActivityTraceFlags.Recorded) != 0 ? SampledValue : NotSampledValue);
+
+            setter(carrier, AWSXRayTraceHeaderKey, sb.ToString());
+        }
+
+        internal static bool TryParseXRayTraceHeader(string rawHeader, out ActivityContext activityContext)
+        {
+            activityContext = default;
```

Comment:
can you put a sample header here, so its easy to visualize whats going on in this method :)

Reply:
Just added